### PR TITLE
New Xcode README.md steps

### PR DESCRIPTION
**Pull request by hrubymar10**
All described steps are accurate. I only ask native speakers to correct my english please. Thank you

### CODE REVIEW

File: projectfiles/Xcode/README.md [hunk 1]
```diff
@@ -30,33 +31,66 @@ Builds for maximum (runtime) speed and compatibility; it builds for 64-bit, with
...
+1. In first chapter we will prepare package for all 3 distribution sources.
+2. In second chapter we will notarize Steam package and we will release it to the SteamStore.
+3. In third chapter we will notarize SourceForge package, we will create `.dmg` image and we will release it to the SourceForge.
+4. In fourth chapter we will release package to the Mac AppStore.
 
+### Packaging - Packages preparation
+ * Obtain sources from github using `git clone -b BRANCH --depth 10 https://github.com/wesnoth/wesnoth /PATH/TO/PROJECT` or use your favourite git program.
+ * Checkout to the latest tag using `cd /PATH/TO/PROJECT ; git checkout TAG` or use your favourite git program.
+ * Use `Finder` to navigate into `/PATH/TO/PROJECT/projectfiles/Xcode`.
+ * Double click on `Fix_Xcode_Dependencies` script and wait until it will be done.
```
Comment (Pentarctagon) [inline projectfiles/Xcode/README.md]: I'd say it as "Double click on **the** `Fix_Xcode_Dependencies` script and wait until it **finishes**."

File: projectfiles/Xcode/README.md [hunk 2]
```diff
@@ -30,33 +31,66 @@ Builds for maximum (runtime) speed and compatibility; it builds for 64-bit, with
...
- * Verify that you signed `.app` propertly using `spctl -a -t exec -vv The\ Battle\ for\ Wesnoth.app`.
- * Rename disk image to match new release version.
- * Unmount it and convert it back using `hdiutil convert /PATH/TO/IMAGE.dmg -format UDBZ -o /PATH/TO/NEW/IMAGE.dmg` command.
- * Sign newly created `.dmg` using `codesign -s "Developer ID Application: Wesnoth, Inc (N5CYW96P9T)" /PATH/TO/NEW/IMAGE.dmg`. You must have Wesnoth's signing certificate.
- * Verify that you signed `.dmg` propertly using `spctl -a -t open --context context:primary-signature -v /PATH/TO/NEW/IMAGE.dmg`.
- * Create SHA-256 checksum using `shasum -a 256 /PATH/TO/NEW/IMAGE.dmg` command.
- * Done! You can release it now.
+ * Update the changelog in `SDLMain.nib` with `changelog.md`.
+ * Create `dist` file using `touch /PATH/TO/PROJECT/data/dist`.
+ * You must increment build number before each distribution package. If you don't know previous build number the best way is probably procceed with steps for one of packages using build number `1` and way with which message notarization step fails. The current build number is contained here.
```
Comment (Pentarctagon) [inline projectfiles/Xcode/README.md]: I'd say it as "You must increment **the** build number before **creating** each distribution package. If you don't know **the** previous build number, the best way is **to proceed** with steps for one of **the** packages using build number `1` and **wait until you get an error** message **that the** notarization step **failed**. The current build number is contained **there**."
(changes bolded)

File: projectfiles/Xcode/README.md [hunk 3]
```diff
@@ -30,33 +31,66 @@ Builds for maximum (runtime) speed and compatibility; it builds for 64-bit, with
...
 
+### Packaging - SteamStore
+ * Find and edit `dist` file in path `/PATH/TO/PROJECT/data/dist`. For Mac AppStore it must contain `Steam`.
```
Comment (Pentarctagon) [inline projectfiles/Xcode/README.md]: Should say for Steam, not for the Mac AppStore.

File: projectfiles/Xcode/README.md [hunk 4]
```diff
@@ -30,33 +31,66 @@ Builds for maximum (runtime) speed and compatibility; it builds for 64-bit, with
...
+ * Now you can hit `Product` > `Archive` from the menubar.
+ * After archivation is done, you can select correct archive in Xcode Organizer, click on `Distribute App`, select `Developer ID` and select `Upload`.
+ * Now you must wait. After sucessful notarization you should get notification via Xcode.
+ * Now click on `Distribute App` again, select `Developer ID` and select export location.
+ * Your final package will be saved in location you selected.
+ * You can now continue to releasing using partner steam tools.
+
+### Packaging - SourceForge
+ * Find and edit `dist` file in path `/PATH/TO/PROJECT/data/dist`. For Mac AppStore it must contain `SourceForge`.
```
Comment (Pentarctagon) [inline projectfiles/Xcode/README.md]: Should say for SourceForge, not for the Mac AppStore.

File: projectfiles/Xcode/README.md [hunk 5]
```diff
@@ -30,33 +31,66 @@ Builds for maximum (runtime) speed and compatibility; it builds for 64-bit, with
...
+ * Now you must wait. After sucessful notarization you should get notification via Xcode.
+ * Now click on `Distribute App` again, select `Developer ID` and select export location.
+ * Your final package will be saved in location you selected.
+ * Now copy dmg template from `/PATH/TO/PROJECT/packaging/macOS/Wesnoth_dmg_packaging_template.dmg`
+ * Convert this template to the R/W image using `hdiutil convert /PATH/TO/TEMPLATE.dmg -format UDRW -o /PATH/TO/NEW/RW_IMAGE.dmg` and mount it.
+ * Copy new wesnoth package using `cp -R /PATH/TO/THE/EXPORTED/PACKAGE.app "/Volumes/The Battle for Wesnoth/The Battle for Wesnoth.app"`.
+ * Unmount R/W image and convert it to final ro image using `hdiutil convert /PATH/TO/RW_IMAGE.dmg -format UDBZ -o /PATH/TO/NEW/IMAGE.dmg` command.
+ * Rename disk image to match new release version. Example: `Wesnoth_1.15.2.dmg`
+ * Sign newly created ro image using `codesign -s "Developer ID Application: Wesnoth, Inc (N5CYW96P9T)" /PATH/TO/NEW/IMAGE.dmg`. (You must have Wesnoth's signing certificate.)
+ * Verify that you signed `.dmg` propertly using `spctl -a -t open --context context:primary-signature -v /PATH/TO/NEW/IMAGE.dmg`.
```
Comment (Pentarctagon) [inline projectfiles/Xcode/README.md]: propertly -> properly

File: projectfiles/Xcode/README.md [hunk 6]
```diff
@@ -30,33 +31,66 @@ Builds for maximum (runtime) speed and compatibility; it builds for 64-bit, with
...
+ * Now you have to notarize the whole `.dmg` image. You must have app specific password for your AppleID prepared.
+ * Execute the `xcrun altool --notarize-app -f /PATH/TO/NEW/IMAGE.dmg --primary-bundle-id org.wesnoth.Wesnoth -u YOUR_APPLE_ID_EMAIL -p YOUR_APPLE_ID_APP_SPECIFIC_PASSWORD --asc-provider N5CYW96P9T` and wait. After sucessful execution it should give you RequestUUID.
```
Comment (Pentarctagon) [inline projectfiles/Xcode/README.md]: sucessful -> successful

File: projectfiles/Xcode/README.md [hunk 7]
```diff
@@ -30,33 +31,66 @@ Builds for maximum (runtime) speed and compatibility; it builds for 64-bit, with
...
+ * You can check notarization status using `xcrun altool --notarization-info REQUEST_UUID -u YOUR_APPLE_ID_EMAIL -p YOUR_APPLE_ID_APP_SPECIFIC_PASSWORD` command. Wait until it get `Package Approved` status message.
```
Comment (Pentarctagon) [inline projectfiles/Xcode/README.md]: "Wait until it **returns** `Package Approved` status message."

File: projectfiles/Xcode/README.md [hunk 8]
```diff
@@ -30,33 +31,66 @@ Builds for maximum (runtime) speed and compatibility; it builds for 64-bit, with
...
+ * Create SHA-256 checksum using `shasum -a 256 /PATH/TO/NEW/IMAGE.dmg > Wesnoth_x.x.x.dmg.sha256` command.
+ * Done! You can release it to the SourceForge now.
```
Comment (Pentarctagon) [inline projectfiles/Xcode/README.md]: "the SourceForge" -> "SourceForge"

File: projectfiles/Xcode/README.md [hunk 9]
```diff
@@ -30,33 +31,66 @@ Builds for maximum (runtime) speed and compatibility; it builds for 64-bit, with
...
 
+### Packaging - Mac AppStore
+ * First you have to enable app sandbox for `wesnothd`.
+ * In Xcode window click on `The Battle for Wesnoth` project in the left sidebar.
+ * Select `wesnothd` target.
+ * Select `Signing & Capabilities` from the top bar.
+ * Click `+ Capability` while in the `All` tab.
+ * Select `App Snadbox`.
```
Comment (Pentarctagon) [inline projectfiles/Xcode/README.md]: Snadbox -> Sandbox

File: projectfiles/Xcode/README.md [hunk 10]
```diff
@@ -30,33 +31,66 @@ Builds for maximum (runtime) speed and compatibility; it builds for 64-bit, with
...
+ * And in newly created section check both `Incomming` and `Outgoing` connections.
```
Comment (Pentarctagon) [inline projectfiles/Xcode/README.md]: Incomming -> Incoming

File: projectfiles/Xcode/README.md [hunk 11]
```diff
@@ -30,33 +31,66 @@ Builds for maximum (runtime) speed and compatibility; it builds for 64-bit, with
...
+ * Find and edit `dist` file in path `/PATH/TO/PROJECT/data/dist`. For Mac AppStore it must contain `macOS App Store`.
+ * Now you can hit `Product` > `Archive` from the menubar.
+ * After archivation is done, you can select correct archive in Xcode Organizer, click on `Distribute App`, select `App Store Connect` and proceed with all steps by clicking `Next`.
+ * After sucessful uploading you must go to the https://appstoreconnect.apple.com/ and continue with releasing there.
```
Comment (Pentarctagon) [inline projectfiles/Xcode/README.md]: sucessful -> successful

File: projectfiles/Xcode/README.md [hunk 12]
```diff
@@ -1,4 +1,5 @@
 # Building with Xcode
+This README describes the way, how wesnoth release packages are build for macOS.
```
Comment (Pentarctagon) [inline projectfiles/Xcode/README.md]: "This README describes the way **to create the** wesnoth release packages for macOS."

**Comment by Pentarctagon** [comment]
Added some comments on the wording.

**Comment by hrubymar10** [comment]
Thank you @Pentarctagon I fixed all comments in latest commit. Is it ok now?

**Comment by Pentarctagon** [comment]
I few more I noticed:
In (3) for the Packaging section, "the SourceForge" -> "SourceForge".

In the Packages preparation list, "if not already by the release manager" -> "if not already **done** by the release manager"

From the SourceForge packaging section, "Your final package will be saved in location you selected." -> "Your final package will be saved in **the** location you selected."

---

Aside from those, it looks fine to me.

**Pull request merged**